compiler/rustc_parse/src/parser/expr.rs

```diff
@@ -2322,7 +2322,21 @@ impl<'a> Parser<'a> {
                 None
             };
             let arrow_span = this.token.span;
-            this.expect(&token::FatArrow)?;
+            if let Err(mut err) = this.expect(&token::FatArrow) {
+                // We might have a `=>` -> `=` or `->` typo (issue #89396).
+                if let token::Eq | token::RArrow = this.token.kind {
```

Comment:
There's a thing called `TokenKind::similar_tokens`, could you use it here?
(Ideally it would be integrated to `expect`, so errors like this would be reported automatically.)

Reply:
Fixed; thanks for the review!